src/main/java/net/imglib2/algorithm/morphology/distance/DistanceTransform.java

```diff
@@ -0,0 +1,804 @@
+		final Cursor< RealComposite< T > > s = Views.flatIterable( Views.collapseReal( dim == lastDim ? Views.interval( source, target ) : Views.permute( Views.interval( source, target ), dim, lastDim ) ) ).cursor();
+		final Cursor< RealComposite< U > > t = Views.flatIterable( Views.collapseReal( dim == lastDim ? target : Views.permute( target, dim, lastDim ) ) ).cursor();
+		final RealComposite< LongType > lowerBoundDistanceIndex = Views.collapseReal( createAppropriateOneDimensionalImage( size, new LongType() ) ).randomAccess().get();
+		final RealComposite< DoubleType > envelopeIntersectLocation = Views.collapseReal( createAppropriateOneDimensionalImage( size + 1, new DoubleType() ) ).randomAccess().get();
+
+		while ( s.hasNext() )
+		{
+			final RealComposite< T > sourceComp = s.next();
+			final RealComposite< U > targetComp = t.next();
+			for ( long i = 0; i < size; ++i )
+				tmp.get( i ).set( sourceComp.get( i ).getRealDouble() );
+			transformSingleColumn( tmp, targetComp, lowerBoundDistanceIndex, envelopeIntersectLocation, d, dim, size );
+		}
+	}
+
+	private static < T extends RealType< T >, U extends RealType< U > > void transformAlongDimensionParallel(
+			final RandomAccessible< T > source,
+			final RandomAccessibleInterval< U > target,
+			final Distance d,
+			final int dim,
+			final ExecutorService es,
+			final int nTasks ) throws InterruptedException, ExecutionException
+	{
+		int largestDim = getLargestDimension( Views.hyperSlice( target, dim, target.min( dim ) ) );
+		// ignore dimension along which we calculate transform
+		if ( largestDim >= dim )
+			largestDim += 1;
+		final long size = target.dimension( dim );
+		final long stepPerChunk = Math.max( size / nTasks, 1 );
+
+		final long[] min = Intervals.minAsLongArray( target );
+		final long[] max = Intervals.maxAsLongArray( target );
+
+		final long largestDimMin = target.min( largestDim );
+		final long largestDimMax = target.max( largestDim );
+
+		final ArrayList< Callable< Void > > tasks = new ArrayList<>();
+		for ( long m = largestDimMin, M = largestDimMin + stepPerChunk - 1; m <= largestDimMax; m += stepPerChunk, M += stepPerChunk )
+		{
+			min[ largestDim ] = m;
+			max[ largestDim ] = Math.min( M, largestDimMax );
+			final Interval fi = new FinalInterval( min, max );
+			tasks.add( () -> {
+				transformAlongDimension( source, Views.interval( target, fi ), d, dim );
+				return null;
+			} );
+		}
+
+		invokeAllAndWait( es, tasks );
+	}
+
+	private static < T extends RealType< T >, U extends RealType< U > > void transformSingleColumn(
+			final RealComposite< T > source,
+			final RealComposite< U > target,
+			final RealComposite< LongType > lowerBoundDistanceIndex,
+			final RealComposite< DoubleType > envelopeIntersectLocation,
+			final Distance d,
+			final int dim,
+			final long size )
+	{
+		long k = 0;
+
+		lowerBoundDistanceIndex.get( 0 ).set( 0 );
+		envelopeIntersectLocation.get( 0 ).set( -1e20 );
+		envelopeIntersectLocation.get( 1 ).set( +1e20 );
```

Comment:
Is there any reason not to use `Double.NEGATIVE_INFINITY` and `Double.POSITIVE_INFINITY` here? (I tried and at least the test passes.)